woocommerce/woocommerce
# Data: reduxify suggested products

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR does the redux implementations of the suggested products endpoint. Relevant changes:

* The store now supports [Thunks](https://developer.wordpress.org/block-editor/how-to-guides/thunks/)
* Add the selector, resolver, and action to handle the storage.

Closes # .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to any page where the store is registered. (Home page, new Product Editor page, etc. )
2. Ensure you have a few products to test
3. Open the dev console
4. Select the suggested products by passing categories, tags, limit. Something like the following:

```es6
await wp.data.resolveSelect( 'wc/admin/products' )
    .getSuggestedProducts( {
        categories: [ 21, 30 ],
        tags: [ 1, 2 ],
        limit: 5,
    } )
```

5 Confirm you see the suggested products.
6. Open the Network tab
7. Confirm that requesting with the same parameters doesn't hit the backend
8. Change the params (categories, tags, limit)
9. Request the suggestions again
10. Confirm it hist the endpoint but only once
11. Repear the request
12. Confirm it doesn't request
13. Additionally, you can take a look at the store by using the Redux dev tool

🍿


https://github.com/woocommerce/woocommerce/assets/77539/13071b83-ecf6-427a-8fe9-b6265560efe1




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
